master/static/migrations/20240407235703_oss-to-ee-catch-up.tx.up.sql

```diff
@@ -0,0 +1,565 @@
+    id text NOT NULL,
+    secret text NOT NULL,
+    domain text NOT NULL,
+    name text NOT NULL,
+
+    CONSTRAINT oauth_clients_pkey PRIMARY KEY (id)
+);
+
+-- migrations/20220418102556_add-user-attributes.tx.up.sql
+ALTER TABLE IF EXISTS scim.users ADD COLUMN IF NOT EXISTS raw_attributes jsonb;
+
+-- migrations/20220628065553_dispatcher-state-persistence.tx.up.sql
+CREATE TABLE IF NOT EXISTS resourcemanagers_dispatcher_dispatches (
+    dispatch_id text PRIMARY KEY,
+    resource_id text NOT NULL REFERENCES allocation_resources(resource_id) ON DELETE CASCADE NOT NULL,
+    allocation_id text NOT NULL
+);
+
+-- migrations/20220628070126_add-original-users-to-dispatches.tx.up.sql
+ALTER TABLE IF EXISTS resourcemanagers_dispatcher_dispatches
+    -- Used to cancel the job, since it must be the original user that cancels it.
+    ADD COLUMN IF NOT EXISTS impersonated_user text NOT NULL;
+
+-- migrations/20220915145212_add-rbac.tx.up.sql
+CREATE TABLE IF NOT EXISTS roles (
+    id integer PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY NOT NULL,
+    role_name TEXT NOT NULL UNIQUE,
+    created_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+CREATE TABLE IF NOT EXISTS role_assignment_scopes (
+    id integer PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+scope_workspace_id integer REFERENCES workspaces (id) ON DELETE CASCADE NULL UNIQUE
+);
+
+CREATE TABLE IF NOT EXISTS role_assignments (
+    group_id integer REFERENCES groups (id),
+    role_id integer REFERENCES roles (id) ON DELETE CASCADE,
+    scope_id integer NULL REFERENCES role_assignment_scopes (id) ON DELETE CASCADE,
+
+    UNIQUE (group_id, role_id, scope_id)
+);
+
+CREATE TABLE IF NOT EXISTS permissions (
+    id integer PRIMARY KEY NOT NULL,
+    name TEXT NOT NULL,
+    global_only boolean NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS permission_assignments (
+    permission_id integer REFERENCES permissions (id) ON DELETE CASCADE,
+    role_id integer REFERENCES roles (id) ON DELETE CASCADE,
+
+    PRIMARY KEY (permission_id, role_id)
+);
+
+-- migrations/20220915145217_add-rbac-precanned-roles-and-permissions.tx.up.sql
+-- TODO: consider adding an IF EXISTS to the insert
+INSERT INTO permissions(id, name, global_only) VALUES
+    (2001, 'create experiment', false),
+    (2004, 'update experiment', false),
+    (5002, 'view project', false),
+    (96001, 'update roles', true),
+    (2005, 'update experiment metadata', false),
+    (93001, 'update group', true),
+    (94001, 'create workspace', true),
+    (4004, 'delete workspace', false),
+    (5003, 'update project', false),
+    (5004, 'delete project', false),
+    (6002, 'assign roles', false),
+    (91001, 'administrate user', true),
+    (2002, 'view experiment artifacts', false),
+    (4003, 'update workspace', false),
+    (2003, 'view experiment metadata', false),
+    (2006, 'delete experiment', false),
+    (4002, 'view workspace', false),
+    (5001, 'create project', false)
+    ON CONFLICT DO NOTHING;
```

Comment:
Since you have the `ON CONFLICT DO NOTHING`, is the TODO now irrelevant? Or was that pre-existing?

Reply:
removed!